pkg/snapstore/utils.go

```diff
@@ -179,36 +178,25 @@ func adaptPrefix(snap *brtypes.Snapshot, snapstorePrefix string) string {
 	return snapstorePrefix
 }
 
-// GetSnapstoreSecretHash returns the hash of object store secrets hash
-func GetSnapstoreSecretHash(config *brtypes.SnapstoreConfig) (string, error) {
+// GetSnapstoreSecretModTime returns the latest modification timestamp of the access credential files.
+// Returns an error if fetching the timestamp of the access credential files fails.
+func GetSnapstoreSecretModTime(config *brtypes.SnapstoreConfig) (time.Time, error) {
 	switch config.Provider {
 	case brtypes.SnapstoreProviderLocal:
-		return "", nil
+		return time.Time{}, nil
```

Comment:
Semantically, returning `time.Time{}` indicates to the _caller_ that the local provider is being used. 

While using `SnapstoreProviderLocal`, there is no concept of updating the `Secret` to access the `Snapstore` and hence there is no valid time that can be returned to the caller of `GetSnapstoreSecretModTime`.

`false` is returned from inside the conditional check 

> `if !newSecretModTime.After(ssr.lastSecretModTime)`

always for the local provider since `newSecretModTime` and `lastSecretModTime` are always `time.Time{}`. Due to this, the explicit check for `time.IsZero` can be avoided.

Reply:
Returning time.Time{} is not semantically correct as the intention of this function is to get the latest modified time. Ideally all providers should have a common interface and one method in the interface should be to check if a secret rotation is in play. For local provider this would result in false thus avoiding the call to `GetSnapstoreSecretModTime`. 

For now lets leave it as is and we will take care of designing this better in steward.